info.go

```diff
@@ -87,6 +88,8 @@ type ProgramInfo struct {
 	Tag string
 	// Name as supplied by user space at load time.
 	Name string
+	// BTF for the program.
+	btf btf.TypeID
```

Comment:
I think you're confusing the BTF ID and the BTF Type ID here.

* BTF ID: a globally unique ID that identifies a collection of types. Basically the equivalent of `btf.Spec`
* BTF Type ID: a per-collection unique ID that identifies a type in a collection. The equivalent of `btf.Type`.

You probably need to introduce a `btf.ID` type and document what it's all about. The wrapper for `BPF_BTF_GET_FD_BY_ID` then operates on `btf.ID`, etc.

Reply:
I think you're correct.

Reply:
Addressed in 51f3b3a